frontend/src/pages/NewCovidTest.js

```diff
@@ -15,16 +15,24 @@ import {
 } from '@material-ui/core';
 import DateAdapter from '@mui/lab/AdapterMoment';
 import {DateTimePicker, LocalizationProvider} from '@mui/lab';
+import PropTypes from 'prop-types';
+import {AuthContext} from '../context/AuthProvider';
 
-export default function NewCovidTest({setAllCovidTests}) {
-	const initialCovidTest = {
-		testType: null,
-		dateTime: null,
-		result: null,
-	};
+NewCovidTest.propTypes = {
+	setAllCovidTests: PropTypes.func.isRequired,
+	newCovidTest: PropTypes.object.isRequired,
+	setNewCovidTest: PropTypes.func.isRequired,
+	initialCovidTest: PropTypes.object,
+};
 
-	const [newCovidTest, setNewCovidTest] = useState(initialCovidTest);
+export default function NewCovidTest({
+	setAllCovidTests,
+	newCovidTest,
+	setNewCovidTest,
+	initialCovidTest,
+}) {
 	const [showError, setShowError] = useState(false);
+	const {token} = useContext(AuthContext);
 	const history = useHistory();
 
```

Comment:
rendering your fromcontrollabels dynamically would save you many lines of code and make it much easier to add/remove elements.
sth like this: 
const values = ['ANTIGEN_LOLLIPOP', ...]
const labels = ['Antigen test - lollipop', ...]
<RadioGroup>{values.map((v, i) => <FormControlLabel value=v control={<Radio/>} label=labels[i] />)}